namespace.go

```diff
@@ -231,12 +238,48 @@ func (ns *Namespace) newClient(ctx context.Context) (*amqp.Client, error) {
 	return amqp.Dial(ns.getAMQPHostURI(), defaultConnOptions...)
 }
 
-func (ns *Namespace) negotiateClaim(ctx context.Context, client *amqp.Client, entityPath string) error {
+// negotiateClaim performs initial authentication and starts periodic refresh of credentials.
+// the returned func is to cancel() the refresh goroutine.
+func (ns *Namespace) negotiateClaim(ctx context.Context, client *amqp.Client, entityPath string) (func() <-chan struct{}, error) {
 	ctx, span := ns.startSpanFromContext(ctx, "sb.namespace.negotiateClaim")
 	defer span.End()
 
 	audience := ns.getEntityAudience(entityPath)
-	return cbs.NegotiateClaim(ctx, audience, client, ns.TokenProvider)
+	if err := cbs.NegotiateClaim(ctx, audience, client, ns.TokenProvider); err != nil {
+		return nil, err
+	}
+	ns.initRefresh.Do(func() {
+		// start the periodic refresh of credentials
+		refreshCtx, cancel := context.WithCancel(context.Background())
+		exitChan := make(chan struct{})
+		go func() {
+			defer func() {
+				// reset the guard when the refresh goroutine exits
+				ns.initRefresh = sync.Once{}
+				// signal that the refresh goroutine has exited
+				close(exitChan)
+			}()
+			for {
+				select {
+				case <-refreshCtx.Done():
+					return
+				case <-time.After(15 * time.Minute):
```

Comment:
My only question is the reasoning for the 15 minutes? Is this in response to the service team saying that the tokens must be at minimum about 15 min?

Reply:
Correct.

Reply:
Ok, wondering if this may cause a problem with a token that expires during one of these 15 min intervals and then it wont get refreshed. You've probably thought about this, but want to understand what you think about this?